### PR TITLE
Debug build use consistent options.

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes

A Debug build is intended to debug QMCPACK source code rather than a compiler option. For this reason, Debug build should carry compiler options as similar to Release build.
GNU/Clang -ffast-math and NVHPC -fast are applied to Debug build as well. So apply them direct to CMAKE_CXX_FLAGS

## What type(s) of changes does this code introduce?
- Build related changes

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: CMake/ClangCompilers.cmake [hunk 1]
```diff
@@ -66,14 +66,12 @@ if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 10.0)
 endif()
 
 # Set extra optimization specific flags
-set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -fomit-frame-pointer -ffast-math")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -fomit-frame-pointer -ffast-math")
-set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -fomit-frame-pointer -ffast-math")
-set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -fomit-frame-pointer -ffast-math")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fomit-frame-pointer -ffast-math")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fomit-frame-pointer -ffast-math")
```
Comment (PDoakORNL) [inline CMake/ClangCompilers.cmake:70]: -fomit-frame-pointer is no good for debugging.
hence the split.

**Review by PDoakORNL — CHANGES_REQUESTED** [review]
Default Debug build should have -fno-omit-frame-pointer

**Comment by prckent** [comment]
Concurring with Peter on wanting (actually needing) the frame-pointer in the debug build.

**Review by PDoakORNL — APPROVED** [review]

**Comment by PDoakORNL** [comment]
It's unclear to me that we have the best gcc and pgi debug flags but I think the exploration of that belongs in another PR.

**Comment by ye-luo** [comment]
Start testing in-house

**Pull request merged**